### ISSUE TITLE
"poetry run ./python.sh" runs python instead of ./python.sh

### ISSUE THREAD

**Issue by gavin-haynes-grow** [python-poetry/poetry#4182]
- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 20.04.2
- **Poetry version**: 1.1.6
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/gavin-haynes-grow/48982eaa0439e2e0cd616522e4472dcc

## Issue

If you init poetry in a directory and install a runnable module, it seems like there is no way to `poetry run` the script named after the module.

I see this with python and `poetry run python.sh`, as well as pytest and `poetry run pytest.sh`

**Comment by finswimmer** [comment]
Hello @gavin-haynes-grow,

can you please add more information about what you trying to achieve? What are you trying to do exactly, what should happen and what happens instead?

These information makes it much easier for us to help you :)

Thanks.

fin swimmer

**Comment by gavin-haynes-grow** [comment]
Here is a full working example: https://github.com/gavin-haynes-grow/poetry-bug

In short,

Expected:
Running `poetry run ./pytest.sh` should run `./pytest.sh` in a poetry shell.

Actual:
Running `poetry run ./pytest.sh` actually runs `poetry run pytest`

**Comment by finswimmer** [comment]
Thanks. :+1: I can reproduce this issue.

**Comment by mkniewallner** [comment]
I believe this issue can be closed, as it seems to have been fixed in version `1.1.9`.

Tested on `1.1.9`, `1.1.13` and latest version from `master` branch, and all of them run the following script correctly, whereas `1.1.6` didn't:

```bash
# pytest.sh

#!/usr/bin/env bash

echo "Hello"
```

```console
$ poetry run ./pytest.sh
Hello
```
